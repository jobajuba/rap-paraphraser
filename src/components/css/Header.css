h1 {
    font-size: 32px;
    text-align: center;
}

p {
    text-align: justify;
    font-size: 20px;
}
