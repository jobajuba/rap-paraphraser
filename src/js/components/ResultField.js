import React from 'react';
import classNames from 'classnames';

export const ResultField = (props) => {
    const {text = '', baseClass = '', isLoading = false} = props;
    const classes = classNames({
        [baseClass]: true,
        'loading': isLoading,
    });
    return (
        <div id="result" className={classes}>{text}</div>
    );
};
