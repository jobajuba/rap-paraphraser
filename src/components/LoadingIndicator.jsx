import React from 'react';
import './css/LoadingIndicator.css';

export const LoadingIndicator = () => {
    return (
        <div className="loader">
            <svg className="circular-loader" viewBox="25 25 50 50">
                <circle
                    className="loader-path"
                    cx="50"
                    cy="50"
                    r="20"
                    fill="none"
                    stroke="#70c542"
                    strokeWidth="2"
                />
            </svg>
        </div>
    );
};
