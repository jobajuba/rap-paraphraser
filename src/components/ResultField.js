import React from 'react';
import classNames from 'classnames';
import './css/ResultField.css';

export const ResultField = (props) => {
    const {text = '', baseClass = '', isLoading = false} = props;
    const classes = classNames({
        [baseClass]: true,
        'result': true,
        'loading': isLoading,
    });
    return (
        <div className={classes}>{text}</div>
    );
};
