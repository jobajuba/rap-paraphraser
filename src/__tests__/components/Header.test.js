import React from 'react';
import {render} from '@testing-library/react';
import {Header} from '../../components/Header';

describe('Test of Header component', () => {

    test('Header component render', () => {
        const {container} = render(<Header />);
        expect(container).toMatchSnapshot();
    });

});
