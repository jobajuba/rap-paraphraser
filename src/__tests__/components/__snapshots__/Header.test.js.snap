// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Header component Header component render 1`] = `
<div>
  <h1>
    Rap-Paraphraser
  </h1>
  <p>
    Добро пожаловать на сайт для перефразирования текста любого типа (в том числе и одного слова) в семантическом поле русского репа с добавлением реп-сленга. Алгоритм перефразирования основан на использовании нейронной сети Word2Vec, обученной на текстах более, чем 9500 песен русских исполнителей, которая находит близкие по семантике слова в контексте русского репа.
  </p>
</div>
`;
